common/report/class.Report.php

```diff
@@ -165,48 +165,77 @@ public function getData() {
 	public function setData($data = null) {
 	    $this->data = $data;
 	}
-	
-	/**
-	 * returns all success elements
-	 * @return array
-	 */
-	public function getSuccesses() {
+
+    /**
+     * returns all success elements
+     *
+     * @param bool $recursive
+     * @return array
+     */
+	public function getSuccesses($recursive = false)
+    {
+        $iterator = true === $recursive
```

Comment:
The main point of this to specifically use the recursive iterator if only `true` as bool has been passed to the method. All of this because the argument is not type hinted. So if you pass any different values we just fall back for the default value. Of course we could use the shorter version but I want to be strict here to prevent passing any type of values accidentally.

Reply:
Fair point. In that case I agree this is the better option